projects/igniteui-angular/src/lib/grids/grid-public-row.ts

```diff
@@ -0,0 +1,492 @@
+    }
+
+    /**
+     * Needs to be removed !
+     *
+     * @hidden
+     */
+    public get rowData(): any {
+        return this.data;
+    }
+
+    /**
+     * Sets/gets whether the row is pinned.
+     * Default value is `false`.
+     * ```typescript
+     * this.grid.selectedRows[0].pinned = true;
+     * ```
+     */
+    public get pinned(): boolean {
+        return this.grid.isRecordPinned(this.data);
+    }
+
+    public set pinned(val: boolean) {
+        if (val) {
+            this.pin();
+        } else {
+            this.unpin();
+        }
+    }
+
+    /**
+     * Sets/gets whether the row is selected.
+     * Default value is `false`.
+     * ```typescript
+     * row.selected = true;
+     * ```
+     */
+    public get selected(): boolean {
+        return this.grid.selectionService.isRowSelected(this.rowID);
+    }
+
+    public set selected(val: boolean) {
+        if (val) {
+            this.grid.selectionService.selectRowsWithNoEvent([this.rowID]);
+        } else {
+            this.grid.selectionService.deselectRowsWithNoEvent([this.rowID]);
+        }
+        this.grid.cdr.markForCheck();
+    }
+
+    /**
+     * Sets whether this specific row has disabled functionality for editing and row selection.
+     * Default value is `false`.
+     * ```typescript
+     * const isDisabled = row.disabled;
+     * ```
+     */
+    public get disabled(): boolean {
+        return this._disabled;
+    }
+
+    /**
+     * Returns the index of the row in the rows collection.
+     */
+    public index: number;
+    private _disabled = false;
+
+    /**
+     * Returns if the row is in delete state.
+     */
+    public get deleted(): boolean {
+        return this.gridAPI.row_deleted_transaction(this.rowID);
+    }
+
+    /**
+     * Returns if the row is currently in edit mode.
+     */
+    public get inEditMode(): boolean {
+        if (this.grid.rowEditable) {
+            const editRowState = this.grid.crudService.row;
+            return (editRowState && editRowState.id === this.rowID) || false;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Returns the child rows. Always returns null for IgxGridRow.
+     */
+    public get children(): any[] {
+        return null;
+    }
+
+    /**
+     * Returns true if child rows exist. Always return false for IgxGridRow.
+     */
+    public get hasChildren(): boolean {
+        return false;
+    }
+
+    /**
+     * Gets/sets the row expanded state.
+     */
+    public get expanded(): boolean {
+        return this.gridAPI.get_row_expansion_state(this.data);
+    }
+
+    public set expanded(val: boolean) {
+        this.gridAPI.set_row_expansion_state(this.rowID, val);
+    }
+
+    /**
+     * Get a reference to the grid that contains the selected row.
+     */
+    protected get grid(): IgxGridComponent | IgxTreeGridComponent | IgxHierarchicalGridComponent {
+        return this._grid as IgxGridComponent;
+    }

```

Comment:
This seems a tad unnecessary.
You can achieve the same directly in the constructor
```typescript
constructor(protected grid: <TypeInfo>, ...) {}
```